src/Middleware/Csrf.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Middleware;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Security\Random;
+use Yiisoft\Security\TokenMasker;
+use Yiisoft\Yii\Web\Cookie;
+use Yiisoft\Yii\Web\Session\SessionInterface;
+
+final class Csrf implements MiddlewareInterface
+{
+    public const CSRF_HEADER = 'X-CSRF-Token';
+
+    private $name = '_csrf';
```

Comment:
Should there be a way to change this name? If yes, setter is needed. If no, it could be a private constant.

Reply:
Why not `protected`? I'd prefer this for *all* framework properties. They should be easy to extend if you really have to.

Reply:
You should not have to :)

Reply:
But what if you do? You never know what awkward requirements may exist in some projects. See, I've had this situation several times over all the years, that I had to copy a complete framework class file because I had to override a property or method that was defined as `private`. That's the worst of all things that can happen IMO. What exactly is the benefit of `private`? Should I open an issue to discuss this for Yii 3.0?

Reply:
That is maintainability vs inheritance-patching issue. If you're making a method `protected` you can not remove, rename it or change its arguments ever since it would break backwards compatibility thus greatly limiting ability of framework maintainers to improve things in minor releases. Therefore I prefer intentional extension/configuration points instead of making everything part of the API.